### PR TITLE
feat(Button): Add ghost buttons

### PR DESCRIPTION
**Pull request by kosanna**
This PR:
- adds a non-required `ghost` prop to the Button
- adds ghost blue button styles
- adds ghost white button style
- if both transparent and ghost attributes applied, ghost gets ignored
- bug fix for button line height
- triggers unreleased changes from the previous PR for ghost buttons #377 

Example usage:
```
<Button color="white" ghost {...restProps} />
<Button color="blue" ghost {...restProps} />
```

### CODE REVIEW

**Comment by markdalgleish** [comment]
Could you please add the release notes to this PR? :)

File: react/Button/Button.less [hunk 1]
```diff
@@ -104,9 +104,11 @@
 }
 
 .root_isGhost {
-  border: 2px solid;
+  @border-thickness: 2px;
+
+  border: @border-thickness solid;
   font-weight: @sk-bold;
-  line-height: normal;
+  line-height: @touchableTextHeight - 2 * @border-thickness;
```
Comment (markdalgleish) [inline react/Button/Button.less]: For readability, could you add parentheses here? e.g. `touchableTextHeight - (@border-thickness * 2)`

Also, for consistency with CSS naming, could you also call this variable `@border-width`?

**Comment by markdalgleish** [comment]
I feel like a transparent ghost button should also be bold, otherwise it will look out of place next to a regular ghost button. What do you think?

**Comment by kosanna** [comment]
I think that transparrent button is kind of a link and we should not touch it's styles if ghost is supplied. There is no use case for that so far, and no design, so lets leave it for now

**Comment by kosanna** [comment]
Otherwise it looks weird that just to get a bold link you supply ghost. Just style it externally with @sk-bold if needed

**Comment by markdalgleish** [comment]
Fair enough 👍

**Pull request merged**